### PR TITLE
Tidy namespaces

### PR DESCRIPTION
**Pull request by mjcarroll**
Switch to nested namespaces, as they are supported now.

Replaces https://github.com/gazebosim/gz-gui/pull/582

Split into three commits for easy review:

* Main code: https://github.com/gazebosim/gz-gui/pull/590/commits/6b2a501598f8684eb328db6b3c3ed0b83769f954
* Plugins: https://github.com/gazebosim/gz-gui/pull/590/commits/ccdef7452953f229ea75fe9dc545e26a38c164cf
* Whitespace https://github.com/gazebosim/gz-gui/pull/590/commits/b939f2eb1ca3996ed529b139018a1ebc4580af5e

### CODE REVIEW

File: include/gz/gui/Application.hh [hunk 1]
```diff
@@ -37,188 +37,195 @@ namespace tinyxml2
...
-      private: std::unique_ptr<ApplicationPrivate> dataPtr;
-    };
-
-    /// \brief Get current running application, this is a cast of qGuiApp.
-    /// \return Pointer to running application, or nullptr if none is running.
-    GZ_GUI_VISIBLE
-    Application *App();
-  }
-}
+class ApplicationPrivate;
```
Comment (ahcorde) [inline include/gz/gui/Application.hh:42]: can we include these classes instead of this forward declaration ?

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: include/gz/gui/Application.hh (hunk 1 above)
Comment (mjcarroll) [reply include/gz/gui/Application.hh:42]: I believe that at least `ApplicationPrivate` doesn't have a declaration in the public API, so we wouldn't be able to.

**Review by azeey — APPROVED** [review]
One comment which should hopefully fix CI.

File: examples/standalone/custom_drawer/custom_drawer.cc
[the remark this answers is not in the document]
Comment (mjcarroll) [reply examples/standalone/custom_drawer/custom_drawer.cc:1]: Yep, I must have missed that one.

**Review by ahcorde — APPROVED** [review]

**Pull request merged**